### PR TITLE
Removed the return-by-value getter of AST node children

### PR DESCRIPTION
**Pull request by ohm314**
Also got rid of the _cr name ending for those functions. Now children
are always returned as const refs. If we need a copy, we construct it
from the const ref.

### CODE REVIEW

File: src/language/templates/visitors/nmodl_visitor.cpp [hunk 1]
```diff
@@ -63,13 +63,17 @@ using namespace ast;
...
     {% elif node.class_name == node_info.BINARY_EXPRESSION_NODE and child.varname == node_info.BINARY_OPERATOR_NAME %}
-        std::string op = node->get_op_cr().eval();
+        std::string op = node->get_op().eval();
         if(op == "=" || op == "&&" || op == "||" || op == "==")
             op = " " + op + " ";
         printer->add_element(op);
     {% else %}
         {% call guard(child.prefix, child.suffix) %}
+    {% if child.is_pointer_node %}
         node->get_{{ child.varname }}(){{ "->" if child.is_pointer_node else "." }}accept(*this);
```
Comment (cattabiani) [inline src/language/templates/visitors/nmodl_visitor.cpp]: `if child.is_pointer_node` is superfluous.I suggest:
`node->get_{{ child.varname }}()->accept(*this);`

File: src/language/templates/visitors/nmodl_visitor.cpp [hunk 2]
```diff
@@ -63,13 +63,17 @@ using namespace ast;
...
+    {% else %}
+        {{ child.class_name }}(node->get_{{ child.varname }}()){{ "->" if child.is_pointer_node else "." }}accept(*this);
```
Comment (cattabiani) [inline src/language/templates/visitors/nmodl_visitor.cpp]: Same comment as before:
`node->get_{{ child.varname }}().accept(*this);`

File: src/language/templates/visitors/nmodl_visitor.cpp (hunk 1 above)
Comment (cattabiani) [reply src/language/templates/visitors/nmodl_visitor.cpp]: or we remove the line before:

`{% if child.is_pointer_node %}`

Comment (ohm314) [reply src/language/templates/visitors/nmodl_visitor.cpp]: yes, sorry, you're right

**Pull request merged**